contracts/interchain-token-service/src/contract.rs

```diff
@@ -597,10 +595,7 @@ impl InterchainTokenService {
                 );
 
                 // Note: attempt to convert a byte string which doesn't represent a valid Soroban address fails at the Host level
-                let minter = minter
-                    .map(|m| Address::from_xdr(env, &m))
-                    .transpose()
-                    .map_err(|_| ContractError::InvalidMinter)?;
+                let minter = minter.map(|m| Address::from_string_bytes(&m));
 
```

Comment:
Also need to change `interchain_transfer` to encode caller address as a string converted to bytes instead of `XDR`

`source_address: caller.clone().to_xdr(env),`